OpenSim/Common/Exception.h

```diff
@@ -47,7 +47,31 @@
 //=============================================================================
 //=============================================================================
 
-namespace OpenSim { 
+namespace OpenSim {
+
+
+/** Macros to throw OpenSim exceptions. The purpose of these macros is to avoid
+having to provide the first few arguments (which are common to all OpenSim
+exceptions) to the exception constructor in the throw statements. This also
+allows us to add more details (eg class name) later easily.                   
+
+Note -- Extra braces enclosing "if" are to avoid problems when these macros are
+called within if-else statements like:
+\code{.cpp}
```

Comment:
Implementation details should not be part of doxygen comments -- those should just be ordinary comments to be seen by developers looking at the source but not by end users of the code. However, the fact that it is OK to use the macros in an `if` statement without braces is great, externally visible, and relevant to end users; just not how it is achieved technically. You shouldn't have to change the user documentation if you change your mind about the internals. In this case for example there might be reasons to make the macro a statement rather than a block (that used to matter in C and earlier C++ versions -- I'm not sure if it still does).